text/0035-schema-applies-to-unspecified.md

```diff
@@ -0,0 +1,167 @@
+# Change the Semantics `appliesTo` Fields in the Schema
+
+## Related issues and PRs
+
+- Reference Issues: [cedar#351](https://github.com/cedar-policy/cedar/issues/351)
+- Implementation PR(s):
+
+## Timeline
+
+- Started: 2023-10-25
+
+## Summary
+
+| Scenario | Status Quo | Proposal |
+| --------- | -------- | ------- |
+| Specified `appliesTo` but omitted `principalTypes` and/or `resourceTypes` | Authz request with this action can only have an unspecified entity as the principal and/or resource | Same as status quo |
+| `appliesTo.principalTypes: []` AND `appliesTo.resourceTypes: []` | Action cannot be used in authz request | Same as status quo |
+| `appliesTo.principalTypes: []` OR `appliesTo.resourceTypes: []` | Action cannot be used in authz request | ❌ Not allowed |
+| Omitted `appliesTo` | Authz request with this action can only have unspecified entities as the principal and resource | ❌ Not allowed |
+
+## Motivation
+
+In Cedar 2.x and 3.0, if an action's `appliesTo.principalTypes` (or `appliesTo.resourceTypes`) is not given, then it's interpreted as an action that applies to only unspecified principals (or resources).
+If an action's `appliesTo` is not given, then it's interpreted as an action that applies only to unspecified principals _and_ resources.
+An _unspecified_ entity has a type that is distinct from any defined type.
+It is created by passing the `None` option in the principal and/or resource component of a [`Request`](https://docs.rs/cedar-policy/latest/cedar_policy/struct.Request.html).
+
+If an action's `appliesTo.principalTypes` or `appliesTo.resourceTypes` are the empty list, then the action is not expected to be used with any request; it should only be used to group other actions.
+This takes precedent over any other information in the `appliesTo`.
+For example: if `appliesTo.principalTypes` is the empty list, then the validator expects the action to never apply, regardless of whether `appliesTo.resourceTypes` is omitted or set to a particular list of entity types.
+
+The difference between omitting a field vs. setting it to the empty list, and the relationship between the two, has caused confusion for users.
+This RFC proposes to:
+
+1. Require that either both `appliesTo.principalTypes` and `appliesTo.resourceTypes` are empty lists, or neither are.
+2. Disallow a missing `appliesTo` field.
+
+These changes remove some of the confusing behavior of the current implementation while maintaining backwards compatibility.
```

Comment:
Instead of "the current implementation", we could more precisely state "Cedar 2.x and 3.0" or even "Cedar 2.x and 3.x" (here and in other places in this RFC).  The tradeoff is that as this RFC text lives to posterity, we may not keep updating this text with which Cedar versions have the described behavior.  Nonetheless, the additional clarity might be worth it.